packages/react-components/lib/reports/render-selected-component-from-dict.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+interface RenderSelectedComponentProps {
+  selectedKey: string;
+  obj: Record<string, React.ReactElement | null>;
+}
+
+export const RenderSelectedComponentFromDict = (
+  props: RenderSelectedComponentProps,
+): React.ReactElement => {
+  const { obj, selectedKey } = props;
+
+  const component = React.useMemo(() => {
+    if (!Object.prototype.hasOwnProperty.call(obj, selectedKey)) {
+      return null;
+    }
+    if (!obj[selectedKey]) {
+      return null;
+    }
+
+    return obj[selectedKey];
+  }, [selectedKey, obj]);
```

Comment:
This causes a re-render even if the selected component is unchanged

```tsx
const obj = React.useState({ unchanged: component });
React.useEffect(() => {
  setObj((prev) => { ...prev, changed: newComponent });
}, []);
return <RenderSelectedComponent selectedKey="unchanged" obj={obj} />
```

Even though only `changed` is updated, `RenderSelectedComponent` still re-renders because `obj` itself is changed.

Reply:
Yeah, you're right, I'll remove this component. I'll add it directly to the ReportDashboard component.